### DISCUSSION TITLE
Can we use webworkers in clearscript (v8)?

### DISCUSSION THREAD

**Discussion by nextfool** [ClearFoundry/ClearScript#311] [General]
Hi,
So I've tried something like "var myWorker = new Worker("my-worker.js"), and it failed says that Worker is not defined.  Then I knew that I am in big trouble because webworkers are a big part of what I am trying to implement with clearscript.

As this answer https://stackoverflow.com/a/22774105 says, nodejs users are luck to have https://www.npmjs.org/package/webworker-threads ,  So my question is, do we have any similar solutions? @ClearScriptLib 

Thank you very much!

**Comment by ClearScriptLib** [comment]
Hi @nextfool,

ClearScript doesn't implement Web APIs, but you can create multiple script engines and use them to run scripts concurrently.

We're working on a sample that demonstrates a `Worker`-like facility in a multithreaded ClearScript environment.

Stay tuned to this discussion topic!

**Comment by ClearScriptLib** [comment]
Hi @nextfool,

Take a look at our sample project [here](https://github.com/microsoft/ClearScriptWorkerSample).

This sample demonstrates a basic facility that's similar to Web Workers. It simulates a ping-pong game using a main script and two workers representing the players.

It also provides an example of how the C# and JavaScript `async` features can be used in combination. The workers' message loops run entirely in the thread pool without tying up any threads.

Good luck!
